### DISCUSSION TITLE
Why do the types for my formatters all become unknown?

### DISCUSSION THREAD

**Discussion by alkstal** [ivanhofer/typesafe-i18n#661] [Q&A] [answered]
I've written a bunch of formatters in `formatters.ts` and the generated types for them in `i18n-types.ts` all define the params and return values as `unknown`, which then causes type errors in the `formatters.ts` file.

Example:

```
export const initFormatters: FormattersInitializer<Locales, Formatters> = (
    locale: Locales,
) => {
    const formatters: Formatters = {
        bold: (text: string) => `<b>${text}</b>`,
        dateLong: date(locale, {dateStyle: 'long'}),
    };

    return formatters;
};
```
```
export type Formatters = {
	bold: (value: unknown) => unknown
	dateLong: (value: unknown) => unknown
}
```

I'm clueless as to why that is and where I might have gone wrong?

Help much appreciated!

**Comment by ivanhofer** [comment, marked as answer]
See this answer: https://github.com/ivanhofer/typesafe-i18n/discussions/643#discussioncomment-5468669

**Reply by alkstal** [reply]
Thanks for pointing that out.

So I have to add the param type within the string where the formatter is used for the type generation to figure that out.
`Some {text:string|bold}` for instance.

That changes the generated type to `bold: (value: string) => unknown`.
Is it expected that the return types are `unknown` or is there a way to get that typed as well?

**Reply by ivanhofer** [reply]
The Translations do not know what the formatter will eventually return. Therefore it is always `unknown`

**Reply by alkstal** [reply]
Thanks for the clarifications!
